plugins/outputs/http/http.go

```diff
@@ -208,6 +209,12 @@ func (h *HTTP) write(reqBody []byte) error {
 	defer resp.Body.Close()
 
 	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		for _, nonRetryableStatusCode := range h.NonRetryableStatusCodes {
+			if resp.StatusCode == nonRetryableStatusCode {
+				return nil
```

Comment:
Can we please add some logging here that the metrics are lost? Something like
```suggestion
				h.Log.Errorf("Received non-retryable status %v. Metrics are lost.", resp.StatusCode)
				return nil
```
Alternatively, you can create and propagate a special `ErrorNoRetry` error and report the issue in the parent `Write` function.